### PR TITLE
Device Cascading delete

### PR DESCRIPTION
**Pull request by murrant**
With a test and better factories

### CODE REVIEW

File: database/factories/FactoryData.php [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+        'g9983',
+        'aluEpon',
+        'aluEponOnu',
+        'aluEponPhysicalUni',
+        'aluEponLogicalLink',
+        'aluGponOnu',
+        'aluGponPhysicalUni',
+        'vmwareNicTeam'
+    ];
+}
```
Comment (laf) [inline database/factories/FactoryData.php]: You're missing a couple :)

**Comment by laf** [comment]
Travis issue on this one.

File: database/factories/FactoryData.php (hunk 1 above)
Comment (murrant) [inline database/factories/FactoryData.php]: haha, I pulled that list from the web.

**Comment by murrant** [comment]
Yeah, I'm trying to figure out why the file is getting pulled in multiple times.  Something to do with the autoloader, which is outside of my area of understanding at the moment.  I might just try something else.

**Pull request merged**